# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/VideoBridgeChild.h
# Commit: d449e8c96753
# Full Hash: d449e8c96753f50a34d3aaf260a00390f729954c
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-06-27 09:34:48
# Regressor Bug: 1561178
# File Overlap Count: 1
# Description:
#   Bug 1561178 - Use Endpoints for VideoBridge construction, and support the possibility that there are multiple at the same time. r=jya
#   
#   In the future we're going to want VideoBridge connections from the RDD process into both the parent process and the GPU process.
#   This does the preparation work for unifying the way we create VideoBridges (using Endpoints, required for cross-process connections),
#   and detects which one to use based on where the video will be composited.
# ==============================================================================

diff -r 557f55a90c8a -r d449e8c96753 gfx/layers/ipc/VideoBridgeChild.h
--- a/gfx/layers/ipc/VideoBridgeChild.h	Thu Jun 27 01:28:56 2019 +0000
+++ b/gfx/layers/ipc/VideoBridgeChild.h	Thu Jun 27 01:29:15 2019 +0000
@@ -19,10 +19,11 @@
  public:
   NS_INLINE_DECL_THREADSAFE_REFCOUNTING(VideoBridgeChild, override);
 
-  static void Startup();
+  static void StartupForGPUProcess();
   static void Shutdown();
 
-  static VideoBridgeChild* GetSingleton();
+  static VideoBridgeChild* GetSingletonToParentProcess();
+  static VideoBridgeChild* GetSingletonToGPUProcess();
 
   // PVideoBridgeChild
   PTextureChild* AllocPTextureChild(const SurfaceDescriptor& aSharedData,
@@ -64,6 +65,8 @@
 
   bool CanSend() { return mCanSend; }
 
+  static void OpenToGPUProcess(Endpoint<PVideoBridgeChild>&& aEndpoint);
+
  private:
   VideoBridgeChild();
   virtual ~VideoBridgeChild();